code/modules/vehicles/mecha/mech_fabricator.dm

```diff
@@ -136,15 +139,19 @@
 	. = ..()
 	if(in_range(user, src) || isobserver(user))
 		. += span_notice("The status display reads: Storing up to <b>[rmat.local_size]</b> material units.<br>Material consumption at <b>[component_coeff*100]%</b>.<br>Build time reduced by <b>[100-time_coeff*100]%</b>.")
-	if(panel_open)
-		. += span_notice("Alt-click to rotate the output direction.")
+		. += span_notice("Currently configured to drop printed objects <b>[dir2text(drop_direction)]</b>.")
 
-/obj/machinery/mecha_part_fabricator/click_alt(mob/user)
-	if(!panel_open)
-		return CLICK_ACTION_BLOCKING
-	dir = turn(dir, -90)
-	balloon_alert(user, "rotated to [dir2text(dir)].")
-	return CLICK_ACTION_SUCCESS
+/obj/machinery/mecha_part_fabricator/mouse_drop_dragged(atom/over, mob/user, src_location, over_location, params)
+	if(!can_interact(user) || (!HAS_SILICON_ACCESS(user) && !isAdminGhostAI(user)) && !Adjacent(user))
```

Comment:
Since you copied this code from autolathe you need to copy it's `interaction_flags_mouse_drag` flags as well else these checks won't work